tests/netperf_test.py

```diff
@@ -65,24 +70,37 @@ def run_netperf_parse(self, device, timeout=60):
     def run_netperf(self, device, ip, opts="", timeout=60):
         self.run_netperf_cmd(device, ip, opts=opts)
         return self.run_netperf_parse(device)
+   
+    def kill_netserver(self, device):
+        device.sendline("kill -9 `pidof netserver`")
+        device.expect(prompt)
 
     def runTest(self):
         super(NetperfTest, self).runTest()
 
         board.sendline('mpstat -P ALL 30 1')
-        speed = self.run_netperf(lan, "192.168.0.1 -c -C -l 30")
-        board.expect('Average.*idle\r\nAverage:\s+all(\s+[0-9]+.[0-9]+){10}\r\n')
+        board.expect('Linux')
+
+	wan_eth1_addr = wan.get_interface_ipaddr(self.iface_dut)
+        speed = self.run_netperf(lan, "%s -c -C -l 30" % wan_eth1_addr)
+
+        board.sendcontrol('c')
+        board.expect('Average.*idle\r\nAverage:\s+all(\s+[0-9]+.[0-9]+){9}\r\n',timeout=60)
         idle_cpu = float(board.match.group(1))
         avg_cpu = 100 - float(idle_cpu)
         lib.common.test_msg("Average cpu usage was %s" % avg_cpu)
+        self.kill_netserver(wan)
 
         self.result_message = "Setup Netperf and Ran Throughput (Speed = %s 10^6bits/sec, CPU = %s)" % (speed, avg_cpu)
 
 def run_netperf_tcp(device, run_time, pkt_size, direction="up"):
+    iface_dut = "eth1"
+    wan_eth1_addr = wan.get_interface_ipaddr(iface_dut)
+    
     if direction == "up":
-        cmd = "netperf -H 192.168.0.1 -c -C -l %s -- -m %s -M %s -D" % (run_time, pkt_size, pkt_size)
+        cmd = "netperf -H %s -c -C -l %s -- -m %s -M %s -D" % (wan_eth1_addr, run_time, pkt_size, pkt_size)
     else:
-        cmd = "netperf -H 192.168.0.1 -c -C -l %s -t TCP_MAERTS -- -m %s -M %s -D" % (run_time, pkt_size, pkt_size)
+        cmd = "netperf -H %s -c -C -l %s -t TCP_MAERTS -- -m %s -M %s -D" % (wan_eth1_addr, run_time, pkt_size, pkt_size)
```

Comment:
I think wan.gw also already has this defined? Not 100% sure.

Reply:
used wan.gw instead of wan.get_interface_ipaddr(self.iface_dut)